### PR TITLE
Fix some Unhandled promise rejections and trim name and surname

### PR DESCRIPTION
**Pull request by chrispanag**
Solving this issue #1

A function trimming the user's first_name and last_name was added on the `Session.ts` file.

Also, some checks have been added to the `setaddr` and `ls` commands, so that they fail gracefully if some necessary steps haven't occured.

### CODE REVIEW

File: dist/classes/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
...
-    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
-}
-Object.defineProperty(exports, "__esModule", { value: true });
-__export(require("./Session"));
+"use strict";
+function __export(m) {
+    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
+}
+Object.defineProperty(exports, "__esModule", { value: true });
+__export(require("./Session"));
```
Comment (kpapadatos) [inline dist/classes/index.js]: @chrispanag You have committed a bunch of files without changes! Only whitespace/LE. Please soft reset your branch and remove them and push again.

Comment (chrispanag) [reply dist/classes/index.js]: Indeed. 
tsc is to blame. Maybe update the `.gitignore` to only take `src` into account (adding `dist` to `.gitignore`)

File: dist/classes/Session.js [hunk 2]
```diff
@@ -1,452 +1,459 @@
...
+        return method.apply(session, arguments);
+    };
+}
+function trimName(user) {
+    console.log(user);
+    user.first_name = user.first_name.trim();
+    user.last_name = user.last_name.trim();
+    console.log(user);
+    return user;
+}
```
[the remark this answers is not in the document]
Comment (chrispanag) [reply dist/classes/Session.js]: Ooops these shouldn't have been there... Maybe I didn't commit that :sweat_smile:

File: dist/cli.js [hunk 3]
```diff
@@ -1,58 +1,66 @@
...
+require('./commands/ls').default(program, session);
+require('./commands/lsaddr').default(program, session);
+require('./commands/lscart').default(program, session);
+require('./commands/menu').default(program, session);
+require('./commands/mkorder').default(program, session);
+require('./commands/setaddr').default(program, session);
+require('./commands/setstore').default(program, session);
+require('./commands/user').default(program, session);
+require('./commands/payment').default(program, session);
+require('./commands/validate').default(program, session);
```
[the remark this answers is not in the document]
Comment (chrispanag) [reply dist/cli.js]: a new version of `tsc` is throwing a bunch of unimportant changes on `dist`. 

As I suggested in my next comment, I think that `.gitignore`-ing the whole `./dist` folder would work the best.

**Comment by chrispanag** [comment]
@kpapadatos friendly bump 😄

File: dist/cli.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kpapadatos) [reply dist/cli.js]: You should use the local version then (`node_modules/.bin/tsc`)! This is an unnecessary change that I don't have time to test 🙁

File: dist/classes/Session.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kpapadatos) [reply dist/classes/Session.js]: Please remove it then :)

File: dist/classes/index.js (hunk 1 above)
Comment (kpapadatos) [reply dist/classes/index.js]: Sure but there's no need for this change now. Please revert these changes and check my other comments and I'd be happy to merge!

**Pull request closed without merging**